subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProjectWrapperRegistry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.project;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DefaultProjectWrapperRegistry implements ProjectWrapperRegistry, HoldsProjectState {
+    private final Map<ProjectIdentity, List<ProjectInternal>> projectWrappers = new ConcurrentHashMap<>();
+
+    @Override
+    public void register(ProjectInternal delegate, ProjectInternal wrapper) {
+        List<ProjectInternal> wrappers = projectWrappers.computeIfAbsent(
+            delegate.getProjectIdentity(),
+            projectIdentity -> Collections.synchronizedList(new ArrayList<>())
```

Comment:
I would expect `CopyOnWriteArrayList` to be better suited here.

Reply:
It will be only writes to this collection. Will it be more suitable for this scenario?